% !TeX program = pdfLaTeX
\documentclass[smallextended]{svjour3}       % onecolumn (second format)
%\documentclass[twocolumn]{svjour3}          % twocolumn
%
\smartqed  % flush right qed marks, e.g. at end of proof
%
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage[utf8]{inputenc}

\usepackage[hyphens]{url} % not crucial - just used below for the URL
\usepackage{hyperref}

%
% \usepackage{mathptmx}      % use Times fonts if available on your TeX system
%
% insert here the call for the packages your document requires
%\usepackage{latexsym}
% etc.
%
% please place your own definitions here and don't use \def but
% \newcommand{}{}
%
% Insert the name of "your journal" with
% \journalname{myjournal}
%

%% load any required packages here


% Pandoc syntax highlighting
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}

% tightlist command for lists without linebreak
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

% From pandoc table feature
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}

% Pandoc citation processing
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
% for Pandoc 2.8 to 2.10.1
\newenvironment{cslreferences}%
  {}%
  {\par}
% For Pandoc 2.11+
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\usepackage{booktabs}
\begin{document}


\title{My Example Computed Manuscript }
 \subtitle{Created in Rmarkdown} 

    \titlerunning{Example computed manuscript}

\author{  Jeffrey M. Perkel \and  }


\institute{
        Jeffrey M. Perkel \at
     Springer Nature, 1 New York Plaza, New York, NY \\
     \email{\href{mailto:jeffrey.perkel@nature.com}{\nolinkurl{jeffrey.perkel@nature.com}}}  %  \\
%             \emph{Present address:} of F. Author  %  if needed
    \and
    }

\date{Received: date / Accepted: date}
% The correct dates will be entered by the editor


\maketitle

\begin{abstract}
A mock computed manuscript created in RStudio using \{Rmarkdown\}. The \{Bookdown\} and \{Rticles\} packages were used to output the text in Springer Nature's desired manuscript format.
\\
\keywords{
    }


\end{abstract}


\def\spacingset#1{\renewcommand{\baselinestretch}%
{#1}\small\normalsize} \spacingset{1}


\hypertarget{intro}{%
\section{Introduction}\label{intro}}

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see \url{http://rmarkdown.rstudio.com}.

Here we'll add some references from Zotero (Perkel 2020): (Fisch et al. 2015; Argelaguet et al. 2021; Lê Cao et al. 2021).

Markdown documents can include inline equations written in \LaTeX, such as \(F = ma\). Here is an equation on its own line:

\[a^2 + b^2 = c^2\]

\hypertarget{results}{%
\section{Results}\label{results}}

\hypertarget{sec:1}{%
\subsection{Inline computation}\label{sec:1}}

One `killer feature' of computed manuscripts is the ability to compute and insert values into the text rather than requiring authors to input them manually. That circumvents the possibility that the author will enter an incorrect number, or forget to update them should new data arise.

For instance, imagine we are analyzing data from a clinical trial. We have grouped subjects in three bins and measured their weights.

\begin{table}

\caption{\label{tab:subject-weights}Original subject data}
\centering
\begin{tabular}[t]{rlrlrlrlrlr}
\toprule
ID & class & wt & | & ID & class & wt & | & ID & class & wt\\
\midrule
1 & Group 2 & 252 & | & 34 & Group 3 & 219 & | & 67 & Group 3 & 132\\
2 & Group 2 & 123 & | & 35 & Group 1 & 218 & | & 68 & Group 3 & 139\\
3 & Group 2 & 174 & | & 36 & Group 1 & 274 & | & 69 & Group 1 & 286\\
4 & Group 2 & 206 & | & 37 & Group 3 & 158 & | & 70 & Group 2 & 103\\
5 & Group 3 & 151 & | & 38 & Group 2 & 216 & | & 71 & Group 3 & 239\\
\addlinespace
6 & Group 1 & 210 & | & 39 & Group 3 & 161 & | & 72 & Group 2 & 129\\
7 & Group 2 & 264 & | & 40 & Group 3 & 103 & | & 73 & Group 1 & 181\\
8 & Group 1 & 222 & | & 41 & Group 3 & 194 & | & 74 & Group 1 & 229\\
9 & Group 3 & 278 & | & 42 & Group 1 & 297 & | & 75 & Group 1 & 260\\
10 & Group 2 & 218 & | & 43 & Group 2 & 132 & | & 76 & Group 1 & 119\\
\addlinespace
11 & Group 1 & 133 & | & 44 & Group 3 & 88 & | & 77 & Group 3 & 199\\
12 & Group 2 & 214 & | & 45 & Group 2 & 127 & | & 78 & Group 2 & 113\\
13 & Group 3 & 126 & | & 46 & Group 2 & 94 & | & 79 & Group 1 & 155\\
14 & Group 1 & 283 & | & 47 & Group 3 & 82 & | & 80 & Group 1 & 180\\
15 & Group 2 & 176 & | & 48 & Group 1 & 179 & | & 81 & Group 3 & 181\\
\addlinespace
16 & Group 2 & 195 & | & 49 & Group 1 & 191 & | & 82 & Group 2 & 123\\
17 & Group 3 & 224 & | & 50 & Group 2 & 147 & | & 83 & Group 3 & 199\\
18 & Group 2 & 90 & | & 51 & Group 2 & 205 & | & 84 & Group 1 & 215\\
19 & Group 1 & 197 & | & 52 & Group 2 & 140 & | & 85 & Group 1 & 216\\
20 & Group 1 & 138 & | & 53 & Group 1 & 253 & | & 86 & Group 3 & 269\\
\addlinespace
21 & Group 1 & 170 & | & 54 & Group 3 & 141 & | & 87 & Group 3 & 249\\
22 & Group 2 & 232 & | & 55 & Group 2 & 245 & | & 88 & Group 3 & 239\\
23 & Group 1 & 251 & | & 56 & Group 3 & 116 & | & 89 & Group 2 & 200\\
24 & Group 3 & 239 & | & 57 & Group 3 & 213 & | & 90 & Group 1 & 191\\
25 & Group 3 & 248 & | & 58 & Group 1 & 120 & | & 91 & Group 1 & 212\\
\addlinespace
26 & Group 2 & 92 & | & 59 & Group 2 & 197 & | & 92 & Group 3 & 110\\
27 & Group 1 & 123 & | & 60 & Group 3 & 224 & | & 93 & Group 1 & 86\\
28 & Group 1 & 90 & | & 61 & Group 2 & 115 & | & 94 & Group 1 & 173\\
29 & Group 2 & 148 & | & 62 & Group 2 & 197 & | & 95 & Group 3 & 211\\
30 & Group 3 & 294 & | & 63 & Group 1 & 291 & | & 96 & Group 2 & 133\\
\addlinespace
31 & Group 2 & 219 & | & 64 & Group 3 & 214 & | & 97 & Group 3 & 165\\
32 & Group 1 & 298 & | & 65 & Group 3 & 281 & | & 98 & Group 3 & 144\\
33 & Group 1 & 222 & | & 66 & Group 2 & 205 & | & 99 & Group 1 & 280\\
\bottomrule
\end{tabular}
\end{table}

Rather than analyzing those data and then copying the results into our manuscript, we can use the programming language \texttt{R} to do that in the manuscript itself. Simply enclose the code inside backticks: \texttt{\textasciigrave{}r\ \textless{}code\ to\ evaluate\textgreater{}\textasciigrave{}}. For instance, to calculate the circumference and area of a circle with radius \emph{r} = 10, you could write ``A = \texttt{\textasciigrave{}r\ pi\ *\ r\^{}2\textasciigrave{}}'' and ``C = \texttt{\textasciigrave{}r\ 2\ *\ pi\ *\ r\textasciigrave{}}. Those give A = 314.159 and C = 62.832.

We have \textbf{99} subjects in our study (Table \ref{tab:subject-weights}). The average weight is \textbf{187.85} (range: \textbf{82}-\textbf{298}). We have \textbf{34} subjects in Group 1, \textbf{32} subjects in Group 2, and \textbf{33} in Group 3. (The numbers in \textbf{bold face type} are computed values.)

Now suppose we get another tranche of data in the form of a second CSV file:

\begin{table}

\caption{\label{tab:new-data}New subject data}
\centering
\begin{tabular}[t]{rlrlrlrlrlr}
\toprule
ID & class & wt & | & ID & class & wt & | & ID & class & wt\\
\midrule
1 & Group 3 & 136 & | & 21 & Group 3 & 197 & | & 41 & Group 2 & 247\\
2 & Group 3 & 227 & | & 22 & Group 2 & 98 & | & 42 & Group 3 & 173\\
3 & Group 3 & 102 & | & 23 & Group 2 & 296 & | & 43 & Group 1 & 236\\
4 & Group 2 & 255 & | & 24 & Group 3 & 90 & | & 44 & Group 1 & 240\\
5 & Group 1 & 191 & | & 25 & Group 1 & 96 & | & 45 & Group 1 & 216\\
\addlinespace
6 & Group 2 & 194 & | & 26 & Group 3 & 124 & | & 46 & Group 3 & 117\\
7 & Group 3 & 246 & | & 27 & Group 2 & 90 & | & 47 & Group 2 & 202\\
8 & Group 3 & 280 & | & 28 & Group 2 & 293 & | & 48 & Group 3 & 220\\
9 & Group 3 & 134 & | & 29 & Group 2 & 259 & | & 49 & Group 1 & 191\\
10 & Group 2 & 76 & | & 30 & Group 2 & 200 & | & 50 & Group 3 & 129\\
\addlinespace
11 & Group 2 & 154 & | & 31 & Group 2 & 169 & | & 51 & Group 3 & 198\\
12 & Group 3 & 197 & | & 32 & Group 2 & 138 & | & 52 & Group 2 & 97\\
13 & Group 1 & 141 & | & 33 & Group 2 & 224 & | & 53 & Group 1 & 281\\
14 & Group 1 & 110 & | & 34 & Group 2 & 155 & | & 54 & Group 3 & 241\\
15 & Group 3 & 216 & | & 35 & Group 1 & 81 & | & 55 & Group 2 & 209\\
\addlinespace
16 & Group 2 & 148 & | & 36 & Group 3 & 216 & | & 56 & Group 1 & 151\\
17 & Group 2 & 188 & | & 37 & Group 2 & 91 & | & 57 & Group 2 & 160\\
18 & Group 3 & 119 & | & 38 & Group 1 & 270 & | & 58 & Group 2 & 216\\
19 & Group 3 & 199 & | & 39 & Group 1 & 256 & | & 59 & Group 1 & 269\\
20 & Group 2 & 124 & | & 40 & Group 3 & 219 & | & 60 & Group 3 & 139\\
\bottomrule
\end{tabular}
\end{table}

There are \textbf{60} subjects in this new dataset (Table \ref{tab:new-data}). Their average weight is \textbf{182.18} (range: \textbf{76}-\textbf{296}).

Combining the two datasets, we have a total of \textbf{159} subjects. The revised average weight is \textbf{185.71} (range: \textbf{76}-\textbf{298}). We now have \textbf{48} subjects in Group 1, \textbf{56} subjects in Group 2, and \textbf{55} in Group 3.

\hypertarget{sec:2}{%
\subsection{Plotting the data}\label{sec:2}}

As Rmarkdown documents can do anything R can do, we can also create and include figures. Here we plot boxplots of our clinical trial data. The data are shown in Figure \ref{fig:plot-data}. Note that this figure number (as well as the table numbers above) are automatically generated.

\begin{figure}
\centering
\includegraphics{SN_templates_files/figure-latex/plot-data-1.pdf}
\caption{\label{fig:plot-data}Weight distribution of clinical trial subjects}
\end{figure}

\hypertarget{methods}{%
\section{Methods}\label{methods}}

The following code was used in section \ref{sec:1} to read in the original clinical trial data from CSV files (these data were simulated):

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# read in some example data}
\NormalTok{df }\OtherTok{\textless{}{-}} \FunctionTok{read\_csv}\NormalTok{(}\StringTok{"example{-}data{-}1.csv"}\NormalTok{)}

\CommentTok{\# display the table, splitting the 99 rows into 3 cols wide}
\NormalTok{tmp }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(df[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{33}\NormalTok{,], }\FunctionTok{rep}\NormalTok{(}\StringTok{\textquotesingle{}|\textquotesingle{}}\NormalTok{, }\DecValTok{33}\NormalTok{), }
\NormalTok{             df[}\DecValTok{34}\SpecialCharTok{:}\DecValTok{66}\NormalTok{,], }\FunctionTok{rep}\NormalTok{(}\StringTok{\textquotesingle{}|\textquotesingle{}}\NormalTok{, }\DecValTok{33}\NormalTok{), }
\NormalTok{             df[}\DecValTok{67}\SpecialCharTok{:}\DecValTok{99}\NormalTok{,])}
\FunctionTok{names}\NormalTok{(tmp) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{\textquotesingle{}ID\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}class\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}wt\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}|\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}ID\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}class\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}wt\textquotesingle{}}\NormalTok{, }
                \StringTok{\textquotesingle{}|\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}ID\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}class\textquotesingle{}}\NormalTok{, }\StringTok{\textquotesingle{}wt\textquotesingle{}}\NormalTok{)}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(tmp, }\AttributeTok{format =} \StringTok{\textquotesingle{}latex\textquotesingle{}}\NormalTok{, }\AttributeTok{booktabs =} \ConstantTok{TRUE}\NormalTok{, }
             \AttributeTok{caption =} \StringTok{"Original subject data"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

The following code was used in section \ref{sec:2} to plot the data:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{final\_data }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{ggplot}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ class, }\AttributeTok{y =}\NormalTok{ wt, }\AttributeTok{color =}\NormalTok{ class)) }\SpecialCharTok{+}
  \FunctionTok{geom\_boxplot}\NormalTok{() }\SpecialCharTok{+}
\NormalTok{  ggbeeswarm}\SpecialCharTok{::}\FunctionTok{geom\_quasirandom}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.25}\NormalTok{) }\SpecialCharTok{+} 
  \FunctionTok{xlab}\NormalTok{(}\StringTok{""}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ylab}\NormalTok{(}\StringTok{"Weight"}\NormalTok{) }\SpecialCharTok{+} 
  \FunctionTok{theme\_minimal}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-argelaguet_computational_2021}{}}%
Argelaguet, Ricard, Anna S. E. Cuomo, Oliver Stegle, and John C. Marioni. 2021. {``Computational Principles and Challenges in Single-Cell Data Integration.''} \emph{Nature Biotechnology}, May, 1--14. \url{https://doi.org/10.1038/s41587-021-00895-7}.

\leavevmode\vadjust pre{\hypertarget{ref-fisch_omics_2015}{}}%
Fisch, K. M., T. Meissner, L. Gioia, J.-C. Ducom, T. M. Carland, S. Loguercio, and A. I. Su. 2015. {``Omics {Pipe}: A Community-Based Framework for Reproducible Multi-Omics Data Analysis.''} \emph{Bioinformatics} 31 (11): 1724--28. \url{https://doi.org/10.1093/bioinformatics/btv061}.

\leavevmode\vadjust pre{\hypertarget{ref-le_cao_community-wide_2021}{}}%
Lê Cao, Kim-Anh, Al J. Abadi, Emily F. Davis-Marcisak, Lauren Hsu, Arshi Arora, Alexis Coullomb, Atul Deshpande, et al. 2021. {``Community-Wide Hackathons to Identify Central Themes in Single-Cell Multi-Omics.''} \emph{Genome Biology} 22 (1): 220. \url{https://doi.org/10.1186/s13059-021-02433-9}.

\leavevmode\vadjust pre{\hypertarget{ref-perkel2020}{}}%
Perkel, Jeffrey M. 2020. {``Streamline Your Writing {\textemdash} and Collaborations {\textemdash} with These Reference Managers.''} \emph{Nature} 585 (7823): 149--50. \url{https://doi.org/10.1038/d41586-020-02491-2}.

\end{CSLReferences}


\bibliographystyle{spbasic}
\bibliography{bibliography.bib}


\end{document}
